src/compiler/crystal/tools/playground/public/session.js

```diff
@@ -297,7 +297,8 @@ Playground.Session = function(options) {
   this.editor._playgroundSession = this;
 
   this.connect = function() {
-    this.ws = new WebSocket("ws://" + location.host + "/client");
+    var socketProtocol = location.protocol === "https:" ? "wss:" : "ws:";
+    this.ws = new WebSocket(socketProtocol + "//" + location.host + "/client");
```

Comment:
We could also take `location.port` into account when constructing the web socket URI, which would then support serving a playground with a non-default port?

Reply:
I'm not certain it is necessary to consult `location.port`.  [`location.host`][0] appears to include the port which is why the socket connection is successful when running `crystal play` defaulting to port **8080**.  Have I missed a different scenario/use case?

[0]: https://developer.mozilla.org/en-US/docs/Web/API/HTMLHyperlinkElementUtils/host

Reply:
My bad, I confused `location.host` with `location.hostname`...please ignore me...